src/sentry/models/notificationsettingbase.py

```diff
@@ -0,0 +1,91 @@
+import sentry_sdk
+from django.conf import settings
+from django.db import models
+
+from sentry.db.models import (
+    BoundedBigIntegerField,
+    BoundedPositiveIntegerField,
+    FlexibleForeignKey,
+    Model,
+)
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.notifications.types import (
+    NotificationScopeType,
+    NotificationSettingOptionValues,
+    NotificationSettingTypes,
+    get_notification_scope_name,
+    get_notification_setting_type_name,
+    get_notification_setting_value_name,
+)
+
+
+class NotificationSettingBase(Model):
+    __include_in_export__ = False
+
+    @property
+    def scope_str(self) -> str:
+        return get_notification_scope_name(self.scope_type)
+
+    @property
+    def type_str(self) -> str:
+        return get_notification_setting_type_name(self.type)
+
+    @property
+    def value_str(self) -> str:
+        return get_notification_setting_value_name(self.value)
+
+    scope_type = BoundedPositiveIntegerField(
+        choices=(
+            (NotificationScopeType.USER, "user"),
+            (NotificationScopeType.ORGANIZATION, "organization"),
+            (NotificationScopeType.PROJECT, "project"),
+            (NotificationScopeType.TEAM, "team"),
+        ),
+        null=False,
+    )
+    scope_identifier = BoundedBigIntegerField(null=False)
+    team_id = HybridCloudForeignKey("sentry.Team", null=True, db_index=True, on_delete="CASCADE")
```

Comment:
what are we doing with team_id?

Reply:
@scttcper if the setting points to a team then we set the `team_id`